src/dash-playlist-loader.js

```diff
@@ -700,110 +681,30 @@ export default class DashPlaylistLoader extends EventTarget {
    */
   refreshXml_() {
-    // The srcUrl here *may* need to pass through handleManifestsRedirects when
-    // sidx is implemented
-    this.request = this.vhs_.xhr({
-      uri: this.masterPlaylistLoader_.srcUrl,
-      withCredentials: this.withCredentials
-    }, (error, req) => {
-      // disposed
-      if (!this.request) {
+    this.requestMaster_((req, masterChanged) => {
+      if (!masterChanged) {
         return;
       }
 
-      // clear the loader's request reference
-      this.request = null;
-
-      if (error) {
-        this.error = {
-          status: req.status,
-          message: 'DASH playlist request error at URL: ' + this.masterPlaylistLoader_.srcUrl,
-          responseText: req.responseText,
-          // MEDIA_ERR_NETWORK
-          code: 2
-        };
-        if (this.state === 'HAVE_NOTHING') {
-          this.started = false;
-        }
-        return this.trigger('error');
-      }
-
-      // xml is the same do nothing.
-      if (req.responseText === this.masterPlaylistLoader_.masterXml_) {
-        return;
+      if (this.media_) {
+        this.media_ = this.master.playlists[this.media_.id];
       }
 
-      this.masterPlaylistLoader_.masterXml_ = req.responseText;
-      this.handleMaster_();
-
       // This will filter out updated sidx info from the mapping
       this.masterPlaylistLoader_.sidxMapping_ = filterChangedSidxMappings(
         this.masterPlaylistLoader_.master,
         this.masterPlaylistLoader_.sidxMapping_
       );
-      const currentSidxInfo = this.media().sidx;
-
-      if (this.media_) {
-        this.media_ = this.master.playlists[this.media_.id];
-      }
-
-      this.updateMinimumUpdatePeriodTimeout_();
-      // current sidx not updated
-      if (!currentSidxInfo) {
-        return;
-      }
-      const sidxKey = generateSidxKey(currentSidxInfo);
-
-      // sidxKey already exists
-      if (this.masterPlaylistLoader_.sidxMapping_[sidxKey]) {
-        return;
-      }
-      const playlist = this.media();
-
-      this.request = requestSidx_(
-        this,
-        playlist.sidx,
-        playlist,
-        this.vhs_.xhr,
-        { handleManifestRedirects: this.handleManifestRedirects },
-        this.sidxRequestFinished_(playlist, this.masterPlaylistLoader_.master, this.state, (newMaster, sidx) => {
-          if (!newMaster || !sidx) {
-            throw new Error('failed to request sidx on minimumUpdatePeriod');
-          }
-
-          // update loader's sidxMapping with parsed sidx box
-          this.masterPlaylistLoader_.sidxMapping_[sidxKey].sidx = sidx;
-
-          addSidxSegmentsToPlaylist(playlist, sidx, playlist.sidx.resolvedUri);
-
-          this.updateMinimumUpdatePeriodTimeout_();
-
-          // TODO: do we need to reload the current playlist?
-          this.refreshMedia_(this.media().id);
-
-          return;
-        })
-      );
 
+      this.addSidxSegments_(this.media(), this.state, (sidxChanged) => {
```

Comment:
We don't have to check `this.media()` has sidx as `addSidxSegments_` will do that now, and the callback will be called.